module/sound/SpeechRecognition/src/SpeechRecognition.cpp

```diff
@@ -0,0 +1,39 @@
+#include "SpeechRecognition.hpp"
+
+#include <vosk_api.h>
+
+#include "extension/Configuration.hpp"
+
+#include "message/input/AudioData.hpp"
+
+namespace module::sound {
+
+    using extension::Configuration;
+    using message::input::AudioData;
+
+
+    SpeechRecognition::SpeechRecognition(std::unique_ptr<NUClear::Environment> environment)
+        : Reactor(std::move(environment)), config{} {
+
+        on<Configuration>("SpeechRecognition.yaml").then([this](const Configuration& cfg) {
+            // Use configuration here from file SpeechRecognition.yaml
+            this->log_level = cfg["log_level"].as<NUClear::LogLevel>();
+        });
+
+        // This reactor receives audiodata from emission
+        on<Trigger<AudioData>>().then([this](const AudioData& audioData) {
+            VoskModel* model           = vosk_model_new("/home/nubots/NUbots/model");
```

Comment:
Move the model to somewhere in the data folder `module/sound/SpeechRecognition/data`.

Reply:
also make the path to the model a config value

Reply:
And move the line into the on configuration, then make the model a class variable.